### PR TITLE
Add an endpoint for validation

### PR DESCRIPTION
**Pull request by loganhz**
https://github.com/rancher/rancher/issues/9608
https://github.com/rancher/rancher/issues/9607

### CODE REVIEW

**Comment by loganhz** [comment]
@yasker FYI

File: validate.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"net/http"
+)
+
+var (
+	validatePort = ":1000"
+	checkOnInit  = true
+	driver       *volumeplugin.RancherStorageDriver
+)
+
+func startValidation(d *volumeplugin.RancherStorageDriver) {
```
Comment (yasker) [inline validate.go:15]: This comment is for NFS part of validation. It's kind of annoying if you're going to use mount to check the connectivity of NFS everytime, even it's already mounted to another directory. Is there a better way?

File: validate.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	driver = d
+	http.HandleFunc("/validate", validate)
+	log.Infof("Listening for validation on 0.0.0.0%v/validate", validatePort)
+	log.Fatal(http.ListenAndServe(validatePort, nil))
+}
+
+func validate(w http.ResponseWriter, r *http.Request) {
+	if checkOnInit {
```
Comment (yasker) [inline validate.go:23]: Why is this needed? We already have healthcheck initialize time.

File: validate.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		checkOnInit = false
+		http.Error(w, "First validation failed as expected", http.StatusInternalServerError)
+		return
+	}
+	err := driver.Validate()
```
Comment (yasker) [inline validate.go:28]: Pattern. `if err:=xxx; err != nil`

File: main.go [hunk 4]
```diff
@@ -115,6 +115,7 @@ func start(c *cli.Context) error {
 			logrus.Fatalf("Error while running healthcheck [%v]", err)
 		}()
 	}
+	go startValidation(d)
```
Comment (yasker) [inline main.go:118]: We already have healthcheck, see the line above.

**Review by yasker — CHANGES_REQUESTED** [review]
